### PR TITLE
Fix issue with hyphen in header

### LINKED ISSUE #49: hyphen present in header PLACE_HOLDER breaks interpolation
### Summary
Using hyphen "-" inside the header which is to be used as PLACE_HOLDER doesn't substitute the PLACE_HOLDER with the  value present in the header of the request.
### Steps To Reproduce

1. Select any route and add this plugin
2. In the headers put **test-header**
3. In the host, replace PLACE_HOLDER by **test-header**
4. Send a request with **test-header** in the header of the request

### PR DESCRIPTION
**Pull request by krishna-raj007**
# Summary
Fixed issue with hyphen in Placeholder in host.

### Issue Description
The plugin's handler.lua used gsub method to replace placeholder in host with corresponding header value. 
gsub does pattern matching and '-' is a special character. '-' matches with 0 or more occurrences of previous character class ([Link](https://www.lua.org/pil/20.2.html)).
Solution: added escape character(%) before '-' wherever it occurs in host.

### Full changelog

* Added a line to replace '-' with '%-' in host
* Removed port which was hardcoded in handler.lua and added is a value in config with default value 80
* Added test cases for both the fixes

### Issues resolved

Fix #49

### CODE REVIEW

File: spec/host-interpolate-by-header/run_spec.lua [hunk 1]
```diff
@@ -277,13 +281,164 @@ for _, strategy in helpers.each_strategy() do
...
+                [host_placeholder] = "abc_xyz",
+              },
+              data = {}
+            }
+          ))
+
+          it(
+            "\nStatus code should be 200",
+            function()
+              assert(res.status == 200)
```
Comment (dhruvd11) [inline spec/host-interpolate-by-header/run_spec.lua:437]: assert for host also to check whether request is proxied to correct URL or not

Comment (krishna-raj007) [reply spec/host-interpolate-by-header/run_spec.lua:439]: Done

File: kong/plugins/host-interpolate-by-header/schema.lua [hunk 2]
```diff
@@ -39,6 +39,12 @@ return {
               required = true
             }
           },
+          {
+            port = typedefs.port{
+              default = 80
+            }
+          },
+
```
Comment (chirag-manwani) [inline kong/plugins/host-interpolate-by-header/schema.lua:47]:
```suggestion
```

File: spec/host-interpolate-by-header/run_spec.lua [hunk 3]
```diff
@@ -9,6 +12,34 @@ for _, strategy in helpers.each_strategy() do
       local bp
       local db
 
+      local function update_config(plugin,host,fallback_host,operation, modulo_by, header_list, port)
```
Comment (chirag-manwani) [inline spec/host-interpolate-by-header/run_spec.lua:15]:
```suggestion
      local function update_config(plugin, host, fallback_host, operation, modulo_by, header_list, port)
```

**Review by dhruvd11 — APPROVED** [review]

**Pull request merged**